static/js/main.js

```diff
@@ -1,69 +1,96 @@
-// Refresh image every 30s
 $('#driver').fileupload({
-      add: function (e, data) {
-        $('#driver-button').prop('disabled', 'disabled');
-          data.submit();
-      },
-      done: function (e, data) {
-        console.log(data);
-        if (data.result['reboot']) {
+  add: function (e, data) {
+    $('#driver-button').prop('disabled', 'disabled');
+    data.submit();
+  },
+  done: function (e, data) {
+    console.log(data);
+    if (data.result['reboot']) {
       if(confirm('In order to fully enable this change, you must reboot the photoframe. Do you wish to do this now?')) {
         $.ajax({
           url:"/maintenance/reboot"
         });
         rebootWatch();
       } else {
-              alert("Upload complete, refreshing drivers");
-              location.reload();
-      }
-      } else {
-            alert("Upload complete, refreshing drivers");
-            location.reload();
-      }
-      },
-      fail: function (e, data) {
-        alert('Failed to upload the driver');
-      },
-      always: function(e, data) {
-        $('#driver-button').prop('disabled', '');
+        alert("Upload complete, refreshing drivers");
+        location.reload();
       }
-  });
+    } else {
+      alert("Upload complete, refreshing drivers");
+      location.reload();
+    }
+  },
+  fail: function (e, data) {
+    alert('Failed to upload the driver');
+  },
+  always: function(e, data) {
+    $('#driver-button').prop('disabled', '');
+  }
+});
 
 $("#driver-button").click(function() {
   $('#driver').trigger('click');
 });
 
 function rebootWatch() {
   $(document.body).html('<h1>Rebooting</h1>')
+  setTimeout(document.location.reload.bind(document.location), 60000);
```

Comment:
Not foolproof, especially if running a slow overworked RPiZW with shitty SD but certainly nicer than nothing 😸